Fix compile warnings when building with pedantic flags

dschwen opened the pull request:
## Purpose

Cleans up the code to build without warnings when specifying pedantic warnings (such as unused variables, wrong initializer list order, potentially mislieading bracketing around logic operators etc.)

**This PR is not complete, but before I sink too much time into it I'd like some quick feedback on whether these changes are welcome or not**

## Author(s)

Daniel Schwen (INL)

## Backward Compatibility

There should not me any functional changes in this PR. It should be backward compatible.

## Implementation Notes

Unused function parameters were `/*param*/` commented out. The names are thus still in the source for self-documentation purposes, but the unused parameter warnings get squashed. Some brackets within chained `&&` terms were shifted to squash logic warnings.

## Post Submission Checklist

_Please check the fields below as they are completed_
- [ ] The feature or features in this pull request is complete
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
- [ ] The source code follows the LAMMPS formatting guidelines

dschwen commented:
The [`kokkos_omp` test failure](https://ci.lammps.org/job/lammps/job/pull-requests/job/kokkos_omp/1072/console) looks unrelated to the changes in this PR.

akohlmey commented:
On Thu, Aug 23, 2018 at 12:30 PM Daniel Schwen <notifications@github.com>
wrote:

> The kokkos_omp test failure
> <https://ci.lammps.org/job/lammps/job/pull-requests/job/kokkos_omp/1072/console>
> looks unrelated to the changes in this PR.
>
yes. @rbberger is working on updating the containers running the tests and
something must have gone unexpectedly wrong.




> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/lammps/lammps/pull/1077#issuecomment-415480946>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AARp4xEuVgmHcEBh99sietZKrq9vj2jYks5uTtiRgaJpZM4WJ2r0>
> .
>


-- 
Dr. Axel Kohlmeyer  akohlmey@gmail.com  http://goo.gl/1wk0
College of Science & Technology, Temple University, Philadelphia PA, USA
International Centre for Theoretical Physics, Trieste. Italy.

rbberger commented:
@dschwen The kokkos_omp failure was caused by me switching the build environment and not updating that particular jenkins job.

dschwen commented:
In the meantime I've written a small python script that fixes the unused parameters automatically :-). So I could finish up this PRs work any time.

rbberger commented:
@dschwen as axel said, this isn't our first rodeo in this regard. the problem isn't fixing warnings. Before putting too much effort into it you need to get @sjplimp on board. Otherwise all this work is for nothing (been there multiple times).

dschwen commented:
Yeah, @rbberger, not much work now that I have automated it, and I certainly didn't want to imply that you don't have any rodeo experience...

![image](https://user-images.githubusercontent.com/202302/44546200-73bf6200-a6d4-11e8-8f5a-8694eef2431c.png)

rbberger commented:
@dschwen we had some private deliberations with @sjplimp  and it looks like he's open to these changes. Please go ahead. Let's take advantage of this window of opportunity.

dschwen commented:
The pr as it is has three changes to core and the granular package. I can
make the changes on the remaining packages tomorrow.

On Thu, Aug 23, 2018, 7:13 PM Richard Berger <notifications@github.com>
wrote:

> @dschwen <https://github.com/dschwen> we had some private deliberations
> with @sjplimp <https://github.com/sjplimp> and it looks like he's open to
> these changes. Please go ahead. Let's take advantage of this window of
> opportunity.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/lammps/lammps/pull/1077#issuecomment-415620550>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAMWPjSIClRS4d-qDFAJXhJxTpLy6gPaks5uT1NSgaJpZM4WJ2r0>
> .
>

dschwen commented:
Update: I have tackles unused parameters in the bulk of the packages (see package status below). A few packages have quite elaborate prerequisites so I have disabled them for now. I'm waiting for the CI test status to see if this needs fixes and will perform another pass looking at the remaining warning types.

<summary>Package status (click arrow to display list)</summary>
<details>

```
Installed YES: package ASPHERE
Installed YES: package BODY
Installed YES: package CLASS2
Installed YES: package COLLOID
Installed YES: package COMPRESS
Installed YES: package CORESHELL
Installed YES: package DIPOLE
Installed  NO: package GPU
Installed YES: package GRANULAR
Installed  NO: package KIM
Installed  NO: package KOKKOS
Installed YES: package KSPACE
Installed YES: package LATTE
Installed YES: package MANYBODY
Installed YES: package MC
Installed YES: package MEAM
Installed YES: package MISC
Installed YES: package MOLECULE
Installed YES: package MPIIO
Installed  NO: package MSCG
Installed YES: package OPT
Installed YES: package PERI
Installed  NO: package POEMS
Installed  NO: package PYTHON
Installed YES: package QEQ
Installed  NO: package REAX
Installed YES: package REPLICA
Installed YES: package RIGID
Installed YES: package SHOCK
Installed YES: package SNAP
Installed YES: package SPIN
Installed YES: package SRD
Installed  NO: package VORONOI
Installed  NO: package USER-ATC
Installed  NO: package USER-AWPMD
Installed YES: package USER-BOCS
Installed YES: package USER-CGDNA
Installed YES: package USER-CGSDK
Installed  NO: package USER-COLVARS
Installed YES: package USER-DIFFRACTION
Installed YES: package USER-DPD
Installed YES: package USER-DRUDE
Installed YES: package USER-EFF
Installed YES: package USER-FEP
Installed  NO: package USER-H5MD
Installed YES: package USER-INTEL
Installed YES: package USER-LB
Installed YES: package USER-MANIFOLD
Installed YES: package USER-MEAMC
Installed YES: package USER-MESO
Installed YES: package USER-MGPT
Installed YES: package USER-MISC
Installed YES: package USER-MOFFF
Installed  NO: package USER-MOLFILE
Installed YES: package USER-NETCDF
Installed  NO: package USER-OMP
Installed YES: package USER-PHONON
Installed  NO: package USER-QMMM
Installed YES: package USER-QTB
Installed YES: package USER-QUIP
Installed YES: package USER-REAXC
Installed YES: package USER-SMD
Installed YES: package USER-SMTBQ
Installed YES: package USER-SPH
Installed YES: package USER-TALLY
Installed YES: package USER-UEF
Installed YES: package USER-VTK
```

</details>

dschwen commented on src/USER-INTEL/fix_intel.cpp:
```diff
@@ -724,11 +724,13 @@ void FixIntel::add_oresults(const ft * _noalias const f_in,
       else
         error->all(FLERR,
                    "Sphere particles not yet supported for gayberne/intel");
+    }
```
dangling else warning

dschwen commented:
Ok, I think I'm done with this for now. The last commit addresses a bunch of unused variable warnings. I had to be very careful as some variable use depends on conditional compilation (preprocessor symbols). 

If variables were used in commented out code I commented out the variable declaration and moved it closer to the commented out code. If variables were used conditionally I moved them into the ifdef blocks. Some variables declarations had elaborate assignments and I preserved them as comments.

I'm not touching the INTEL module code any further as that seems to be  a minefield (variables used in  pragmas etc.) I cannot test that part properly and it looks like neither does the CI.
